java/dagger/lint/DaggerKotlinIssueDetector.kt

```diff
@@ -200,7 +200,7 @@ class DaggerKotlinIssueDetector : Detector(), SourceCodeScanner {
       override fun visitMethod(node: UMethod) {
         if (!node.isConstructor &&
           node.hasAnnotation(PROVIDES_ANNOTATION) &&
-          node.hasAnnotation(JVM_STATIC_ANNOTATION)
```

Comment:
Note that `UMethod` doesn't `override` `hasAnnotation` and delegates it to the underlying `javaPsi` (of `PsiMethod`). So,  having `node.hasAnnotation(...) == true` means the underlying light-class method has `@JvmStatic` that comes from the source PSI.

Changing like this will avoid the crash below, but cause false negatives: a `static` (e.g. (`companion`) `object` functions) with `@JvmStatic` won't be caught.

---

The right solution is to change:

```kt
val annotation = node.findAnnotation(JVM_STATIC_ANNOTATION)!!
```

to

```kt
val annotation = node.findAnnotation(JVM_STATIC_ANNOTATION)
  ?: node.javaPsi.modifierList.findAnnotation(JVM_STATIC_ANNOTATION)
```

where the first line still searches for `UAnnotation` for `@JvmStatic` in `UMethod`, and as a fall back for already-static method, the second line searches for the LC modeling of `@JvmStatic` in `PsiMethod`.